### PR TITLE
Added utility and changed how vision updates pose/odometry

### PR DESCRIPTION
**Pull request by witcpalek**
When disabled used MegaTag1's rotation
Don't update when rotating fast
Changed the initial rotation angle for MegaTag2

### CODE REVIEW

File: src/main/cpp/chassis/SwerveChassis.h [hunk 1]
```diff
@@ -122,6 +122,7 @@ class SwerveChassis : public IChassis, public LoggableItem
     ISwerveDriveState *GetSpecifiedDriveState(ChassisOptionEnums::DriveStateType driveOption);
 
     bool IsRotating() const { return m_isRotating; }
+    double GetRotationRateDegreesPerSecond() const { return m_pigeon != nullptr ? m_pigeon->GetRate() : 0.0; }
```
Comment (Tan560) [inline src/main/cpp/chassis/SwerveChassis.h:125]: You didn't use this in SwerveChassis.cpp

Line 239 is still this:
 auto useVision = (m_pigeon != nullptr && std::abs(m_pigeon->GetRate()) < 720.0);

File: src/main/cpp/chassis/SwerveChassis.cpp [hunk 2]
```diff
@@ -290,18 +290,22 @@ void SwerveChassis::UpdateOdometry()
...
     {
-        std::optional<VisionPose> megaTag2Pose = m_vision->GetRobotPositionMegaTag2(GetYaw(), // mtAngle.Degrees(),
-                                                                                    units::angular_velocity::degrees_per_second_t(0.0),
-                                                                                    units::angle::degree_t(0.0),
-                                                                                    units::angular_velocity::degrees_per_second_t(0.0),
-                                                                                    units::angle::degree_t(0.0),
-                                                                                    units::angular_velocity::degrees_per_second_t(0.0));
-
-        if (megaTag2Pose)
+        auto useVision = (m_pigeon != nullptr && std::abs(m_pigeon->GetRate()) < 720.0);
```
Comment (Tan560) [inline src/main/cpp/chassis/SwerveChassis.cpp:293]: See other comment

**Review by Tan560 — CHANGES_REQUESTED** [review]

File: src/main/cpp/chassis/SwerveChassis.h (hunk 1 above)
Comment (witcpalek) [reply src/main/cpp/chassis/SwerveChassis.h:125]: Updated here and a couple of other places to use the new call

File: src/main/cpp/chassis/SwerveChassis.cpp (hunk 2 above)
Comment (witcpalek) [reply src/main/cpp/chassis/SwerveChassis.cpp]: Updated call

**Review by Tan560 — APPROVED** [review]

File: src/main/cpp/DragonVision/DragonVision.cpp [hunk 3]
```diff
@@ -555,6 +555,7 @@ std::optional<VisionPose> DragonVision::GetRobotPositionMegaTag2(units::angle::d
 		LimelightHelpers::SetRobotOrientation(m_dragonCameraMap[RobotElementNames::CAMERA_USAGE::LAUNCHE]->GetCameraName(),
 											  yaw.value(),
 											  0, 0, 0, 0, 0);
+		Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, std::string("Pose"), std::string("Limelight Yaw"), yaw.value());
```
Comment (Tan560) [inline src/main/cpp/DragonVision/DragonVision.cpp:558]: Remove Logging

File: src/main/cpp/auton/drivePrimitives/ResetPositionPathPlanner.cpp [hunk 4]
```diff
@@ -47,25 +48,47 @@ void ResetPositionPathPlanner::Init(PrimitiveParams *param)
...
-            if (FMSData::GetInstance()->GetAllianceColor() == frc::DriverStation::Alliance::kRed)
+            auto initialPose = path.get()->getPreviewStartingHolonomicPose();
+
+            auto vision = DragonVision::GetDragonVision();
+
+            auto visionPosition = vision->GetRobotPosition();
+            auto hasVisionPose = visionPosition.has_value();
+            auto initialRot = hasVisionPose ? visionPosition.value().estimatedPose.ToPose2d().Rotation().Degrees() : initialPose.Rotation().Degrees();
+
+            Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, string("Pose"), string("Path Rotation"), initialPose.Rotation().Degrees().value());
```
Comment (Tan560) [inline src/main/cpp/auton/drivePrimitives/ResetPositionPathPlanner.cpp:62]: Remove Logging

File: src/main/cpp/auton/drivePrimitives/ResetPositionPathPlanner.cpp [hunk 5]
```diff
@@ -47,25 +48,47 @@ void ResetPositionPathPlanner::Init(PrimitiveParams *param)
...
+            auto hasMegaTag2Position = megaTag2Position.has_value() && std::abs(chassis->GetRotationRateDegreesPerSecond()) < m_maxAngularVelocityDegreesPerSecond;
+
+            if (hasMegaTag2Position)
             {
-                path = path.get()->flipPath();
+                auto pose = megaTag2Position.value().estimatedPose.ToPose2d();
+
+                ResetPose(pose);
+                Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, string("Pose"), string("Mega2 Reset Pose Rotation"), pose.Rotation().Degrees().value());
+                Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, string("Pose"), string("Mega2 Reset Pose X"), pose.X().value());
```
Comment (Tan560) [inline src/main/cpp/auton/drivePrimitives/ResetPositionPathPlanner.cpp:81]: Remove Logging

**Review by Tan560 — COMMENTED** [review]
Commit DragonLimelight changes from Monday night and remove logging

File: src/main/cpp/auton/drivePrimitives/ResetPositionPathPlanner.cpp (hunk 4 above)
Comment (witcpalek) [reply src/main/cpp/auton/drivePrimitives/ResetPositionPathPlanner.cpp]: tanay updated

File: src/main/cpp/auton/drivePrimitives/ResetPositionPathPlanner.cpp (hunk 5 above)
Comment (witcpalek) [reply src/main/cpp/auton/drivePrimitives/ResetPositionPathPlanner.cpp]: tanay updated

**Pull request merged**